htdocs/fourn/ajax/getSupplierPrices.php

```diff
@@ -86,12 +86,46 @@
 
 	// Add price for costprice
 	$price=$producttmp->cost_price;
+	if (empty($price) && ! empty($conf->global->PRODUCT_USE_SUB_COST_PRICES_IF_COST_PRICE_EMPTY))
+	{
+		// get costprice for subproducts if any
+		$producttmp->get_sousproduits_arbo();
+		$prods_arbo=$producttmp->get_arbo_each_prod();
+		if (!empty($prods_arbo))
+		{
+			$price = 0;
+			foreach($prods_arbo as $child)
+			{
+				$sousprod = new Product($db);
+				$sousprod->fetch($child['id']);
+				$price += $sousprod->cost_price;
+			}
+		}
+	}
+
 	$prices[] = array("id" => 'costprice', "price" => price2num($price), "label" => $langs->trans("CostPrice").': '.price($price, 0, $langs, 0, 0, -1, $conf->currency), "title" => $langs->trans("PMPValueShort").': '.price($price, 0, $langs, 0, 0, -1, $conf->currency));  // For price field, we must use price2num(), for label or title, price()
 
 	if(!empty($conf->stock->enabled))
 	{
 		// Add price for pmp
 		$price=$producttmp->pmp;
+		if (empty($price))
```

Comment:
May be the test on PRODUCT_USE_SUB_COST_PRICES_IF_COST_PRICE_EMPTY must be added here too ?

Reply:
@eldy : i am not sure that we want pmp to be calculated on components... maybe another constant dedicated to pmp ?
By the way, i think there is a bug on the calculation because quantity should be multiplied to subproduct pmp (and costprice), dont you think so ?
i dont touch anything because i really dont understand exactly how the get_arbo_each_prod function works (for me it returns an array that is never filled...)